src/codegen/codegen-type-definition.lisp

```diff
@@ -127,9 +127,20 @@
 
    (loop :for constructor :in (tc:type-definition-constructors def)
          :for name := (tc:constructor-entry-name constructor)
-         :for ty := (tc:lookup-value-type env name) 
+         :for ty := (tc:lookup-value-type env name)
+         :for docstring := (source:docstring constructor)
          :collect `(setf (documentation ',name 'variable)
-                         ,(format nil "~A :: ~A" name ty))
+                         ,(format nil "~A :: ~A~A"
+                                  name
+                                  ty
+                                  (if docstring
+                                      (format nil "~%~A" docstring)
+                                      "")))
```

Comment:
nit: figure out the nullary condition format string magic character -- or make this a helper function